dhis-2/dhis-support/dhis-support-hibernate/src/test/java/org/hisp/dhis/datasource/DatabasePoolUtilsTest.java

```diff
@@ -0,0 +1,169 @@
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.datasource;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertInstanceOf;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+
+import com.mchange.v2.c3p0.ComboPooledDataSource;
+import com.mchange.v2.c3p0.DriverManagerDataSource;
+import com.zaxxer.hikari.HikariDataSource;
+import java.beans.PropertyVetoException;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.stream.Collectors;
+import javax.sql.DataSource;
+import org.apache.logging.log4j.core.Logger;
+import org.apache.logging.log4j.core.LoggerContext;
+import org.apache.logging.log4j.core.test.appender.ListAppender;
+import org.hisp.dhis.external.conf.ConfigurationKey;
+import org.hisp.dhis.external.conf.DhisConfigurationProvider;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+public class DatabasePoolUtilsTest {
+
+  private static ListAppender appender;
+
+  @BeforeAll
+  public static void beforeAll() {
+    LoggerContext loggerContext = LoggerContext.getContext(false);
+    Logger logger = (Logger) loggerContext.getLogger(DatabasePoolUtils.class);
+    appender = new ListAppender("Test");
+    appender.start();
+    loggerContext.getConfiguration().addLoggerAppender(logger, appender);
+  }
+
+  @BeforeEach
+  public void beforeEach() throws SQLException {
+    appender.clear();
+    StubDriver jdbcDriver = new StubDriver();
+    DriverManager.registerDriver(jdbcDriver);
+  }
+
+  @Test
+  public void testCreateDbPoolWhenDbPoolTypeIsUnPooled()
+      throws PropertyVetoException, SQLException {
+
+    DhisConfigurationProvider mockDhisConfigurationProvider = mock(DhisConfigurationProvider.class);
+    given(mockDhisConfigurationProvider.getProperty(ConfigurationKey.CONNECTION_DRIVER_CLASS))
+        .willReturn("mock");
+
+    DatabasePoolUtils.PoolConfig.PoolConfigBuilder poolConfigBuilder =
+        new DatabasePoolUtils.PoolConfig.PoolConfigBuilder()
+            .dbPoolType(DatabasePoolUtils.DbPoolType.UNPOOLED.name())
+            .jdbcUrl("jdbc:fake:db")
+            .username("")
+            .password("")
+            .dhisConfig(mockDhisConfigurationProvider);
+
+    DataSource dataSource = DatabasePoolUtils.createDbPool(poolConfigBuilder.build());
+    assertInstanceOf(DriverManagerDataSource.class, dataSource);
+
+    assertEquals("Database pool type value is [UNPOOLED]", getLogEntry());
```

Comment:
Instead of asserting on log entries to confirm the chosen pool config, could we assert on the actual DataSource properties returned from those methods?
Each datasource seems to have unique requirements which could be confirmed with assertions. This would leave the test in a much more reliable state, being able to catch actual changes in code (which a log statement cannot)
That would also remove the need for all the logging setup & dependency in this PR.

Reply:
sorry @claudemamo I added a few more comments after pulling the code locally and checking it again (as opposed to browsing the changes in GitHub)

Reply:
that log assertion proves the code path which is fine (it still leaves the datasource properties untested)